ClassCastException, DefaultFilesystemStoreImpl cannot be cast to ReactiveContentStore

**Describe the bug**
ClassCastException is thrown when 'org.springframework.boot:spring-boot-starter-webflux' is included
Spring boot version: 2.6.5
Spring content version: 2.1.0

Exception:
`2022-03-29 20:55:50.213[0;39m [31mERROR[0;39m [35m12724[0;39m [2m---[0;39m [2m[  restartedMain][0;39m [36mo.s.boot.SpringApplication              [0;39m [2m:[0;39m Application run failed

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'profilePictureStore': Invocation of init method failed; nested exception is java.lang.ClassCastException: class internal.org.springframework.content.fs.repository.DefaultFilesystemStoreImpl cannot be cast to class org.springframework.content.commons.repository.ReactiveContentStore (internal.org.springframework.content.fs.repository.DefaultFilesystemStoreImpl and org.springframework.content.commons.repository.ReactiveContentStore are in unnamed module of loader 'app')
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804) ~[spring-beans-5.3.17.jar:5.3.17]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620) ~[spring-beans-5.3.17.jar:5.3.17]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[spring-beans-5.3.17.jar:5.3.17]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.17.jar:5.3.17]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.17.jar:5.3.17]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.17.jar:5.3.17]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.17.jar:5.3.17]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:934) ~[spring-beans-5.3.17.jar:5.3.17]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918) ~[spring-context-5.3.17.jar:5.3.17]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[spring-context-5.3.17.jar:5.3.17]
	at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:64) ~[spring-boot-2.6.5.jar:2.6.5]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:740) ~[spring-boot-2.6.5.jar:2.6.5]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:415) ~[spring-boot-2.6.5.jar:2.6.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) ~[spring-boot-2.6.5.jar:2.6.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1312) ~[spring-boot-2.6.5.jar:2.6.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1301) ~[spring-boot-2.6.5.jar:2.6.5]
	at com.example.demo.DemoApplication.main(DemoApplication.java:10) ~[main/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49) ~[spring-boot-devtools-2.6.5.jar:2.6.5]
Caused by: java.lang.ClassCastException: class internal.org.springframework.content.fs.repository.DefaultFilesystemStoreImpl cannot be cast to class org.springframework.content.commons.repository.ReactiveContentStore (internal.org.springframework.content.fs.repository.DefaultFilesystemStoreImpl and org.springframework.content.commons.repository.ReactiveContentStore are in unnamed module of loader 'app')
	at org.springframework.content.commons.repository.factory.AbstractStoreFactoryBean.createContentStore(AbstractStoreFactoryBean.java:204) ~[spring-content-commons-2.1.0.jar:na]
	at org.springframework.content.commons.repository.factory.AbstractStoreFactoryBean.initAndReturn(AbstractStoreFactoryBean.java:167) ~[spring-content-commons-2.1.0.jar:na]
	at org.springframework.content.commons.repository.factory.AbstractStoreFactoryBean.afterPropertiesSet(AbstractStoreFactoryBean.java:162) ~[spring-content-commons-2.1.0.jar:na]
	at internal.org.springframework.content.fs.config.FilesystemStoreFactoryBean.afterPropertiesSet(FilesystemStoreFactoryBean.java:35) ~[spring-content-fs-2.1.0.jar:na]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863) ~[spring-beans-5.3.17.jar:5.3.17]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800) ~[spring-beans-5.3.17.jar:5.3.17]
	... 21 common frames omitted

` 

**To Reproduce**
Steps to reproduce the behavior:
1. Check out demo: [ClassCast demo](https://github.com/KarstenMathiasen/spring-content-classcast-demo) 
2. Start the project
3. During start it will throw a ClassCastExcpetion
4. See error above

--- a/spring-content-commons/src/main/java/org/springframework/content/commons/repository/factory/AbstractStoreFactoryBean.java
+++ b/spring-content-commons/src/main/java/org/springframework/content/commons/repository/factory/AbstractStoreFactoryBean.java
@@ -10,6 +10,7 @@ import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 
+import org.apache.commons.lang.ClassUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.springframework.aop.framework.ProxyFactory;
@@ -176,7 +177,7 @@ public abstract class AbstractStoreFactoryBean
 		// Create proxy
 		ProxyFactory result = new ProxyFactory();
 		result.setTarget(target);
-        if (!REACTIVE_STORAGE) {
+        if (!ClassUtils.getAllInterfaces(storeInterface).contains(ReactiveContentStore.class)) {
             result.setInterfaces(new Class[] { storeInterface, Store.class, AssociativeStore.class, ContentStore.class, ParameterTypeAware.class });
         } else {
             result.setInterfaces(new Class[] { storeInterface, Store.class, ReactiveContentStore.class, ParameterTypeAware.class });
@@ -198,7 +199,7 @@ public abstract class AbstractStoreFactoryBean
 
 		StoreMethodInterceptor intercepter = new StoreMethodInterceptor();
 
-		if (!REACTIVE_STORAGE) {
+		if (!ClassUtils.getAllInterfaces(storeInterface).contains(ReactiveContentStore.class)) {
 		    storeFragments.add(new StoreFragment(storeInterface, new StoreImpl((ContentStore<Object, Serializable>) target, publisher, Paths.get(System.getProperty("java.io.tmpdir")))));
 		} else {
             storeFragments.add(new StoreFragment(storeInterface, new ReactiveStoreImpl((ReactiveContentStore<Object, Serializable>) target, publisher)));

--- a/spring-content-s3/src/main/java/internal/org/springframework/content/s3/config/S3StoreFactoryBean.java
+++ b/spring-content-s3/src/main/java/internal/org/springframework/content/s3/config/S3StoreFactoryBean.java
@@ -1,10 +1,12 @@
 package internal.org.springframework.content.s3.config;
 
+import org.apache.commons.lang.ClassUtils;
 import org.springframework.aop.framework.ProxyFactory;
 import org.springframework.beans.factory.BeanFactory;
 import org.springframework.beans.factory.NoSuchBeanDefinitionException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.content.commons.repository.ReactiveContentStore;
 import org.springframework.content.commons.repository.factory.AbstractStoreFactoryBean;
 import org.springframework.content.commons.utils.PlacementService;
 import org.springframework.content.s3.config.MultiTenantS3ClientProvider;
@@ -72,7 +74,7 @@ public class S3StoreFactoryBean extends AbstractStoreFactoryBean {
 		DefaultResourceLoader loader = new DefaultResourceLoader();
 		loader.addProtocolResolver(s3Protocol);
 
-		if (!REACTIVE_STORAGE) {
+        if (!ClassUtils.getAllInterfaces(getStoreInterface()).contains(ReactiveContentStore.class)) {
 		    if (client == null) {
 		        throw new NoSuchBeanDefinitionException(S3Client.class.getCanonicalName());
 		    }

